File: db4on/core/src/core/com/db4o/foundation/EmptyIterator.cs
namespace com.db4o.foundation
{
	internal sealed class EmptyIterator : com.db4o.foundation.Iterator4
	{
		internal EmptyIterator()
		{
		}

		public bool MoveNext()
		{
			return false;
		}

		public object Current()
		{
			throw new System.InvalidOperationException();
		}
	}
}

File: db4oj/core/src/com/db4o/foundation/EmptyIterator.java
/* Copyright (C) 2004   db4objects Inc.   http://www.db4o.com */

package com.db4o.foundation;


final class EmptyIterator implements Iterator4 {

	EmptyIterator() {
	}

	public final boolean moveNext() {
		return false;
	}

	public Object current() {
		throw new IllegalStateException();
	}
}